File: ExecutionService/docker/javascript/execute.js
const fs = require('fs');
const { exec } = require('child_process');

function executeNodeCode(code) {
    // Write the input JavaScript code to a file
    fs.writeFileSync('script.js', code);

    // Run the script using Node.js
    exec('node script.js', (error, stdout, stderr) => {
        if (error) {
            console.error(`Error: ${stderr || error.message}`);
            process.exit(1); // Ensure proper exit on failure
        }
        console.log(stdout); // Print the script output
        process.exit(0); // Ensure proper exit on success
    });
}

// Read from stdin
let input = '';
process.stdin.on('data', (chunk) => {
    input += chunk;
});

process.stdin.on('end', () => {
    const code = input.trim();
    if (!code) {
        console.error('Error: No Node.js code provided.');
        process.exit(1);
    }
    executeNodeCode(code);
});

File: ExecutionService/docker/python/execute.py
import sys
import subprocess

def execute_python_code(code):
    try:
        # Write the input code to a temporary file
        with open("script.py", "w") as script_file:
            script_file.write(code)

        # Run the script and capture output
        result = subprocess.run(
            ["python", "script.py"],
            text=True,
            capture_output=True,
            timeout=10  # Set timeout for execution
        )

        # Check for runtime errors
        if result.returncode != 0:
            return f"Error: {result.stderr.strip()}"

        # Return successful output
        return result.stdout.strip()

    except subprocess.TimeoutExpired:
        return "Error: Code execution timed out."
    except Exception as e:
        return f"Error: {str(e)}"

# Main execution
if __name__ == "__main__":
    code = sys.stdin.read().strip()
    if not code:
        print("Error: No Python code provided.")
        sys.exit(1)
    print(execute_python_code(code))
